File: tests/test1.pas
program test1;
var
i : integer;
k : integer;
procedure foo;
var
    j: integer;
begin
    j := 1;
    repeat
        write(j);
        j := j + 1;
    until j = 3;
end;
begin
    foo;
    i := 0;
    while i <> 4 do
    begin
        write(i);
        i := i + 1;
    end;
    for k := 0 to 2 do
    begin
        write('Hello');
    end;
    for k := 4 downto 0 do
    begin
        write('World');
    end;
end.

File: tests/test1.c
#include <stdio.h> 
 
void foo(){
int j;
j = 1;
do { 
printf(j);
j = j + 1;
} while(!(j == 3));
}


int main(void){
int i;
int k;
foo();
i = 0;
while(i != 4) {
printf(i);
i = i + 1;
}
for(k = 0; k <= 2; k++){ 
printf("Hello");
}
for(k = 4; k >= 0; k--){ 
printf("World");

}
return 0; 
}